### DISCUSSION TITLE
Interactive UI Elements?

### DISCUSSION THREAD

**Discussion by david-littlefield** [gildas-lormeau/SingleFile#987] [General]
Is there a way for SingleFile to keep the event listeners for UI elements? 

For example, clicking on the menu icon to display a dropdown menu.

**Comment by gildas-lormeau** [comment]
Yes and no. Most of the time, it won't work because scripts are not designed to be saved for offline reading. You can find more info here: https://github.com/gildas-lormeau/SingleFile/blob/master/faq.md#why-dont-interactive-elements-like-folding-titles-dynamic-maps-or-carousels-work-properly-in-saved-pages

**Comment by david-littlefield** [comment]
Gotcha. That would save all scripts, not just user interface elements, correct?

**Comment by gildas-lormeau** [comment]
Yes, all the scripts would be saved.

**Reply by david-littlefield** [reply]
Would it be realistic to only save js that modifies the user interface? For example, open and close navigation.

**Reply by gildas-lormeau** [reply]
It would maybe have been the case in the early 2000s by saving the `on...` attributes (e.g. `onclick`). Unfortunately (?) while this technique still works, it is rarely used in HTML pages today. It wouldn't probably suffice though because the are some chances that an external JavaScript function would be called and it would already start to become quite painful to manage actually. Of course, web pages today are more complex and define these behaviors only in scripts. And there, it's almost impossible to identify the code required to make an interactive element work.

**Reply by david-littlefield** [reply]
Gotcha. Ok, thanks!
